mmtrack/datasets/got10k_dataset.py

```diff
@@ -37,38 +141,31 @@ def format_results(self, results, resfile_path=None):
         print('-------- There are total {} images --------'.format(
             len(track_bboxes)))
 
-        video_info = self.coco.videos
-        format_results = defaultdict(list)
-        for img_id, track_bbox in enumerate(track_bboxes):
-            img_info = self.data_infos[img_id]
-            assert img_info['id'] == img_id + 1, 'img id is not matched'
-            video_name = video_info[img_info['video_id']]['name']
-            format_results[video_name].append(track_bbox[:4])
-
-        assert len(video_info) == len(
-            format_results
-        ), 'The number of video is not matched! There are {} videos in the \
-            dataset and {} videos in the testing results'.format(
-            len(video_info), len(format_results))
-
-        # writing submitted results
-        # TODO record test time
-        print('writing submitted results to {}'.format(resfile_path))
-        for video_name, bboxes in format_results.items():
-            video_file_path = osp.join(resfile_path, video_name)
-            if not osp.isdir(video_file_path):
-                os.makedirs(video_file_path, exist_ok=True)
-            video_txt = osp.join(video_file_path,
-                                 '{}_001.txt'.format(video_name))
-            with open(video_txt, 'w') as f:
-                for bbox in bboxes:
+        start_ind = end_ind = 0
+        for num, video_info in zip(self.num_frames_per_video, self.data_infos):
+            end_ind += num
+            video_name = video_info['video_path'].split('/')[-1]
+            video_resfiles_path = osp.join(resfile_path, video_name)
+            if not osp.isdir(video_resfiles_path):
+                os.makedirs(video_resfiles_path, exist_ok=True)
+            video_bbox_txt = osp.join(video_resfiles_path,
+                                      '{}_001.txt'.format(video_name))
+            video_time_txt = osp.join(video_resfiles_path,
+                                      '{}_time.txt'.format(video_name))
+            with open(video_bbox_txt,
+                      'w') as f_bbox, open(video_time_txt, 'w') as f_time:
+
+                for bbox in results['track_bboxes'][start_ind:end_ind]:
                     bbox = [
                         str(f'{bbox[0]:.4f}'),
                         str(f'{bbox[1]:.4f}'),
                         str(f'{(bbox[2] - bbox[0]):.4f}'),
                         str(f'{(bbox[3] - bbox[1]):.4f}')
                     ]
                     line = ','.join(bbox) + '\n'
-                    f.writelines(line)
+                    f_bbox.writelines(line)
+                    f_time.writelines('0.04\n')
```

Comment:
remove the time cost, since it has not been tested

Reply:
The time cost must be written in the submitted files, otherwise, the testing server can not work.

Reply:
may be we can set `record_test_time=False` as a parameter in `format_results` function.